src/Lachain.Core/Blockchain/SystemContracts/NativeTokenContract.cs

```diff
@@ -128,6 +147,50 @@ public ExecutionStatus Approve(UInt160 spender, UInt256 amount, SystemContractEx
             frame.ReturnValue = 1.ToUInt256().ToBytes();
             return ExecutionStatus.Ok;
         }
+        
+        [ContractMethod(Lrc20Interface.MethodSetAllowedSupply)]
+        public ExecutionStatus SetAllowedSupply(Money amount, SystemContractExecutionFrame frame)
+        {
+            frame.UseGas(GasMetering.NativeTokenApproveCost);
+            if(!frame.InvocationContext.Sender.Equals(_mintCntrlAdd))
+                return ExecutionStatus.ExecutionHalted;
+            
+            if(amount > _maxSupply)
+                return ExecutionStatus.ExecutionHalted;
+            
+            if(amount <= _allowedSupply)
```

Comment:
No,  there is some misunderstanding here,  it is possible to decrease allowed supply,  but we can't make it less than current **total** supply

Reply:
incorporated